pydantic/generics.py

```diff
@@ -60,11 +60,14 @@ def __class_getitem__(  # type: ignore
             _generic_types_cache[(cls, params[0])] = created_model
         return created_model
 
-
-def concrete_name(cls: Type[Any], params: Tuple[Type[Any], ...]) -> str:
-    param_names = [param.__name__ if hasattr(param, '__name__') else str(param) for param in params]
-    params_component = ', '.join(param_names)
-    return f'{cls.__name__}[{params_component}]'
+    @classmethod
+    def _concrete_name_(cls: Type[Any], params: Tuple[Type[Any], ...]) -> str:
```

Comment:
I used a sunder name to indicate the method should be considered public, despite using a leading underscore to prevent namespace conflicts. (Rather than using only a leading underscore.)

Reply:
please can you use `__concrete_name__` to be consistent with other names.